twitter/app/Models/Tweet.php

```diff
@@ -86,4 +94,36 @@ public function searchByQuery(string $keyword): Collection
             return $searchdTweet;
         }
     }
+
+    /**
+     * いいねしているかを調べる
+     */
+    public function isFavorite(int $userId): bool
+    {
+        return $this->favorites()->where('user_id', $userId)->exists();
+    }
+
+    /**
+     * いいねする
+     */
+    public function favorite(int $userId): void
+    {
+        if($this->isFavorite($userId)) {
+
+        } else {
+            $this->favorites()->attach($userId);
+        }
```

Comment:
`$this->isFavorite($userId)`がtrueだった時の処理は？
ないのであれば、以下のようにしよう
```suggestion
        if(!$this->isFavorite($userId))
            $this->favorites()->attach($userId);
```